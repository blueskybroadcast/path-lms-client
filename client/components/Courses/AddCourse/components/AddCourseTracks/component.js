import React from 'react';
import PropTypes from 'prop-types';

const AddCourseTracks = ({
  tracks, handleTrackChange, handleTrackDelete, handleTrackAdd
}) => (
  <div className="option-list tracks">
    <div className="header">
      <span>Available tracks</span>
      <a
        onClick={handleTrackAdd}
        className="add-option add-track tooltiped"
        data-placement="left"
        data-toggle="tooltip"
        title=""
        data-original-title="Add new track"
      />
    </div>
    <ul>
      { Object.entries(tracks).map(([id, data]) => {
        return !data._destroy ? (
          <li key={id}>
            <input
              value={data.name}
              onChange={({ target }) => handleTrackChange(target.value, id)}
              className="track-name"
              placeholder="Enter name"
              aria-label="Track Name"
              type="text"
              name="course[tracks_attributes][0][name]"
              id="course_tracks_attributes_0_name"
            />
            <div className="track-actions">
              <span>0 items</span>
              <a
                onClick={() => handleTrackDelete(id)}
                className="remove-track"
                href="#"
              >
                <i className="icon icon-times" />
              </a>
            </div>
          </li>
        ) : null;
      })}
    </ul>
  </div>
);

AddCourseTracks.propTypes = {
  tracks: PropTypes.objectOf(PropTypes.object).isRequired,
  handleTrackChange: PropTypes.func.isRequired,
  handleTrackDelete: PropTypes.func.isRequired,
  handleTrackAdd: PropTypes.func.isRequired
};

export default AddCourseTracks;
