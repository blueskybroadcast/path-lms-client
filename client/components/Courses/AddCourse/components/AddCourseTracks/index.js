import AddCourseEditableList from './component';

export default AddCourseEditableList;
