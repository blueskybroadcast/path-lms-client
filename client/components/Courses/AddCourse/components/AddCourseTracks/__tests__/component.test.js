import React from 'react';
import { shallow } from 'enzyme';
import AddCourseTracks from '../component';

describe('AddCourseTracks', () => {
  test('should render AddCourseTracks correctly', () => {
    const tracks = {
      0: {
        _destroy: false,
        name: 'Track 1'
      },
      1: {
        _destroy: false,
        name: 'Track 2'
      }
    };
    const wrapper = shallow(
      <AddCourseTracks
        tracks={tracks}
        handleTrackChange={jest.fn()}
        handleTrackDelete={jest.fn()}
        handleTrackAdd={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
