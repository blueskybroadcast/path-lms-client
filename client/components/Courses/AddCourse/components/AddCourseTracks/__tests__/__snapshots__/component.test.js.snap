// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCourseTracks should render AddCourseTracks correctly 1`] = `
<div
  className="option-list tracks"
>
  <div
    className="header"
  >
    <span>
      Available tracks
    </span>
    <a
      className="add-option add-track tooltiped"
      data-original-title="Add new track"
      data-placement="left"
      data-toggle="tooltip"
      onClick={[MockFunction]}
      title=""
    />
  </div>
  <ul>
    <li
      key="0"
    >
      <input
        aria-label="Track Name"
        className="track-name"
        id="course_tracks_attributes_0_name"
        name="course[tracks_attributes][0][name]"
        onChange={[Function]}
        placeholder="Enter name"
        type="text"
        value="Track 1"
      />
      <div
        className="track-actions"
      >
        <span>
          0 items
        </span>
        <a
          className="remove-track"
          href="#"
          onClick={[Function]}
        >
          <i
            className="icon icon-times"
          />
        </a>
      </div>
    </li>
    <li
      key="1"
    >
      <input
        aria-label="Track Name"
        className="track-name"
        id="course_tracks_attributes_0_name"
        name="course[tracks_attributes][0][name]"
        onChange={[Function]}
        placeholder="Enter name"
        type="text"
        value="Track 2"
      />
      <div
        className="track-actions"
      >
        <span>
          0 items
        </span>
        <a
          className="remove-track"
          href="#"
          onClick={[Function]}
        >
          <i
            className="icon icon-times"
          />
        </a>
      </div>
    </li>
  </ul>
</div>
`;
